securedrop_client/gui/main.py

```diff
@@ -78,15 +79,29 @@ def autosize_window(self):
         screen = QDesktopWidget().screenGeometry()
         self.resize(screen.width(), screen.height())
 
-    def show_login(self, error=None):
+    def show_login(self):
         """
         Show the login form. If an error message is passed in, the login
         form will display this too.
         """
-        self.login_view.reset()
-        if error:
-            self.login_view.error(error)
-        self.main_view.update_view(self.login_view)
+        self.login_dialog = LoginDialog(self)
+        self.login_dialog.setup(self.controller)
+        self.login_dialog.reset()
+        self.login_dialog.exec()
+
+    def show_login_error(self, error):
+        """
+        Display an error in the login dialog.
+        """
+        if self.login_dialog and error:
+            self.login_dialog.error(error)
```

Comment:
hey so when there is an auth error in `logic.py/Client.on_authenticate`, it looks like `LoginDialog` is set as disabled for me:

![screen shot 2018-10-17 at 3 31 17 pm](https://user-images.githubusercontent.com/7832803/47119926-b3f91780-d221-11e8-8ba9-522ace39a239.png)

we could add `self.setDisabled(False)` after the error message is displayed in `LoginDialog.error`?

Reply:
Doh... this will be easy to fix and is something I should have refactored out. Good catch. :-)